.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
```

This can be pinned too (though lower risk, since this is an alias to 24.04, updates are infrequent, and the runners are controlled by GitHub).
```suggestion
    runs-on: ubuntu-24.04
```

---

It doesn't provide a security benefit, but pinning it is a good idea anyway because runner updates have a tendency to randomly break stuff in my experience, thanks for suggesting it.